readme.md

```diff
@@ -0,0 +1,41 @@
+# Field Encrypt
+The Field Encrypt module allows fields to be stored in an encrypted state rather than in plain text and allows these fields to be decrypted when loaded for viewing.
+
+## Documentation
+There is an architecture overview available in /documentation. This is available as an image file as well as a draw.io xml file. This should give a general idea of how each of the portions of the module function and how the module hooks into Drupal and it's plugin system.
+
+## FieldEncryptProcessEntities Service
+The main function of the FieldEncryptProcessEntities Service is to provide a way to process entities.
+
+In Drupal 8, it is recommended that all processing on fields be done at the entity level. As such, the service primarily takes entities as inputs and processes those fields in order to either decrypt encrypted fields or encrypt fields before storage.
+
+The `encrypt_entity()` and `decrypt_entity()` methods perform these actions. Additionally, after we have changed the storage settings (enabled / disabled encryption), we must process existing fields. `encrypt_stored_field()` and `decrypt_stored_field()` provide that functionality.
+
+Inside the service, we iterate over each field and then each of the fields values. For example, the `text_with_summary` field type has a `value` and a `summary` value. The encryption itself is then handled by other services. For example, text encryption is handled by the `encryption` service as part of the `encrypt` module.
```

Comment:
Literally each field?

May be best to do the entity_load_multiple_by_properties with 'encrypt' == '1'